File: 0x14-bit_manipulation/0-binary_to_uint.c
#include "main.h"
/**
 * binary_to_uint - exactly what it says on the tin
 * @binary: a string representing a binary number
 */
unsigned int binary_to_uint(const char *binary)
{
	unsigned int number = 0;
	int index = 0;
	int power = 1;

	if (!binary)
		return (0);

	while (b[index])
	{
		if (binary[index] != '0' && binary[index] != '1')
			return (0);
		index++;
	}
	index--;

	while (index >= 0)
	{
		if (binary[index] == '1')
			number += power;
		power *= 2;
	}

	return (number);
}

File: 0x14-bit_manipulation/0-binary_to_uint.py
#!/usr/bin/env python3

def binary_to_uint(binary):
    number = 0
    mult = 0
    length = 0

    if (binary == None):
        return (0)

    length = len(binary)

default_test_case = [
    "1",
    "101",
    "1e01",
    "1100010",
    "0000000000000000000110010010"
]

for case in default_test_case:
    print(case)